sql/clients_last_seen_v1.sql

```diff
@@ -19,9 +19,26 @@ WITH current_sample AS (
 SELECT
   @submission_date AS submission_date,
   CURRENT_DATETIME() AS generated_time,
+
+  -- This last_date struct is used to record the last date that a particular
+  -- client met various criteria; we record a null date if the client does
+  -- not meet a given criterion.
+  STRUCT (
+    IF(
+      country IN ('US', 'FR', 'DE', 'UK', 'CA'),
+      current_sample.submission_date,
+      previous.last_date.seen_in_tier1_country
+    ) AS seen_in_tier1_country,
+    IF(
+      scalar_parent_browser_engagement_total_uri_count_sum >= 5
+      current_sample.submission_date,
+      previous.last_date.visited_5_uri
+    ) AS visited_5_uri
+  ) AS last_date,
+
   IF(current_sample.client_id IS NOT NULL,
     current_sample,
-    previous).*
+    previous).* EXCEPT (last_date)
```

Comment:
in order for this part to work the struct `current_sample` and `previous` must have the same schema

Reply:
filed #7 to fix this